packages/fast-usdc/src/exos/advancer.js

```diff
@@ -145,10 +149,17 @@ export const prepareAdvancerKit = (
               return;
             }
 
-            const { borrowerFacet, poolAccount } = this.state;
+            const { borrowerFacet, poolAccount, settlementAddress } =
+              this.state;
             const { recipientAddress } = evidence.aux;
             const decoded = decodeAddressHook(recipientAddress);
             mustMatch(decoded, AddressHookShape);
+            if (!settlementAddress?.value) {
+              throw Fail`⚠️ No 'settlementAddress'. must call 'publishAddresses' first.`;
```

Comment:
`publishAddresses` is outside the concern of this module. It only knows about `setSettlementAddress`

since `handleTransactionEvent` will fail if `setSettlementAddress` hasn't been called, maybe the contract shouldn't start the watcher until it has.

but… why not require `settlementAddress` when making the exo? It's known at that time. Then you can rely on it being set and not have to handle mutations.

Reply:
We only have the account at the time the exo is created. Agree it would be better to require it at `make` time. If awaiting `getAddress()` doesn't break the upgrade test, I'm up for it.

Reply:
I gave it a try in ac84a973a70a08df88e9d4816379febdfd8bf779. Let's see how force:integration does with it.

Another reason to make the change is POLA… advancer should not be able to access funds from the settlementAccount.